activerecord/test/cases/primary_keys_test.rb

```diff
@@ -373,6 +373,22 @@ def test_composite_primary_key_out_of_order
     assert_equal ["code", "region"], @connection.primary_keys("barcodes_reverse")
   end
 
+  def test_assigning_a_composite_primary_key
+    book = Cpk::Book.new
+    book.id = [1, 2]
+    book.save!
+
+    assert_equal [1, 2], book.id
```

Comment:
Wonder if it would be useful to additionally assert for every part of the pk like
```ruby
assert_equal(1, book.author_id)
assert_equal(2, book.number)
```